templates/_wrapper.pyx

```diff
@@ -140,11 +140,17 @@ def set_state(state):
     cdef REAL_t[:, :, ::1] input_value_3d
     cdef REAL_t[:, ::1] input_value_2d
     cdef REAL_t[::1] input_value_1d
+    cdef int dt_physics
     tracer_metadata = get_tracer_metadata()
     cdef set processed_names_set = set()
     for name, quantity in state.items():
         if name == 'time':
             set_time(state[name])
+        elif name == SURFACE_PRECIPITATION_RATE:
+            quantity = quantity.transpose([pace.util.Y_DIMS, pace.util.X_DIMS])
+            get_physics_timestep_subroutine(&dt_physics)
+            quantity.view[:] *= dt_physics / MM_PER_M
```

Comment:
I think this will modify the input, since this is a 2D variable can you make it perform this modification on a new copy? I think you can do this with deepcopy, or by making a new quantity and copying the data over.

Reply:
Good call. Done.